src/java/org/apache/cassandra/cql3/statements/schema/CopyTableStatement.java

```diff
@@ -0,0 +1,178 @@
+        return new AuditLogContext(AuditLogEntryType.CREATE_TABLE_LIKE, targetKeyspace, targetTableName);
+    }
+
+    @Override
+    public Keyspaces apply(ClusterMetadata metadata)
+    {
+        Keyspaces schema = metadata.schema.getKeyspaces();
+        KeyspaceMetadata sourceKeyspaceMeta = schema.getNullable(sourceKeyspace);
+        TableMetadata sourceTableMeta = sourceKeyspaceMeta.getTableNullable(sourceTableName);
+
+        if (null == sourceKeyspaceMeta)
+            throw ire("Source Keyspace '%s' doesn't exist", sourceKeyspace);
+
+        if (null == sourceTableMeta)
+            throw ire("Souce Table '%s'.'%s' doesn't exist", sourceKeyspace, sourceTableName);
+
+        if (sourceTableMeta.isIndex())
+            throw ire("Cannot use CREATE TABLE LIKE on a index table '%s'.'%s'.", sourceKeyspace, sourceTableName);
+
+        if (sourceTableMeta.isView())
+            throw ire("Cannot use CREATE TABLE LIKE on a materialized view '%s'.'%s'.", sourceKeyspace, sourceTableName);
+
+        KeyspaceMetadata targetKeyspaceMeta = schema.getNullable(targetKeyspace);
+        if (null == targetKeyspaceMeta)
+            throw ire("Target Keyspace '%s' doesn't exist", targetKeyspace);
+
+        if (targetKeyspaceMeta.hasTable(targetTableName))
+        {
+            if(ifNotExists)
+                return schema;
+
+            throw new AlreadyExistsException(targetKeyspace, targetTableName);
+        }
+        // todo support udt for differenet ks latter
+        if (!sourceKeyspace.equalsIgnoreCase(targetKeyspace) && !sourceKeyspaceMeta.types.isEmpty())
+            throw ire("Cannot use CREATE TABLE LIKE across different keyspace when source table have UDTs.");
+
+        String sourceCQLString = sourceTableMeta.toCqlString(false, false, true, false);
+        // add all user functions to be able to give a good error message to the user if the alter references
+        // a function from another keyspace
+        UserFunctions.Builder ufBuilder = UserFunctions.builder().add();
+        for (KeyspaceMetadata ksm : schema)
+            ufBuilder.add(ksm.userFunctions);
+
+        TableMetadata.Builder targetBuilder = CreateTableStatement.parse(sourceCQLString,
+                                                                         targetKeyspace,
+                                                                         targetTableName,
+                                                                         sourceKeyspaceMeta.types,
+                                                                         ufBuilder.build())
+                                                                  .indexes(Indexes.none())
+                                                                  .triggers(Triggers.none());
+
+        TableParams originalParams = targetBuilder.build().params;
+        TableParams newTableParams = attrs.asAlteredTableParams(originalParams);
+        TableMetadata table = targetBuilder.params(newTableParams)
```

Comment:
@Maxwell-Guo  could you please test what happen when I do this?

````
CREATE TABLE ks.tb_copy LIKE ks.tb WITH id = '....';
````

Can I provide a custom ID from my copied table?

Reply:
This will give you ID too:

````
DESCRIBE TABLE ks.tb3 WITH INTERNALS ;
````

You can then take that "WITH ID", increase it by one and create that "like" table and see what happens.

Reply:
I think we need to find a way how to support this.

Reply:
@blerer double check this one, what we have now is 

````
.id(attrs.hasProperty(TableAttributes.ID) ? attrs.getId() : TableId.get(metadata))
````